samples/algorithms/repeat-until-success/RepeatUntilSuccess.qs

```diff
@@ -0,0 +1,110 @@
+namespace Microsoft.Quantum.Samples.RepeatUntilSuccess {
+    open Microsoft.Quantum.Intrinsic;
+    open Microsoft.Quantum.Canon;
+    open Microsoft.Quantum.Math;
+    open Microsoft.Quantum.Preparation;
+
+    /// Example of a Repeat-until-success circuit
+    /// The circuit is executed on a "target" qubit using an "ancilla" and 
+    /// "resource" qubit.
+    /// The goal is to measure Zero for both the ancilla and resource qubit.
+    /// If this succeeds, the circuit will have effectively applied an 
+    /// Rz(arctan(2)) gate (also known as V_3 gate) on the target qubit.
+    /// If this fails, rerun the circuit up to <limit> times.
+    @EntryPoint()
+    operation ApplyRzArcTan2(
+        inputValue : Bool,
+        inputBasis: Pauli,
+        limit: Int
+    ) : (Bool, Result, Int) {
+        using ((ancilla, resource, target) = (Qubit(), Qubit(), Qubit())) {
+            // Prepare qubits
+            PrepareXZero(ancilla);
+            PrepareXZero(resource);
+            PrepareValueForBasis(inputValue, inputBasis, target);
+
+            // Initialize results to One by default.
+            mutable result1 = One;
+            mutable result2 = One;
```

Comment:
Similarly here, you can avoid making a mutable declaration by bringing `result2` inside the loop.